app/src/main/java/com/duckduckgo/app/tabs/model/TabDataRepository.kt

```diff
@@ -201,7 +203,10 @@ class TabDataRepository @Inject constructor(
     override fun deleteAll() {
         Timber.i("Deleting tabs right now")
         tabsDao.deleteAllTabs()
-        GlobalScope.launch { webViewPreviewPersister.deleteAll() }
+        GlobalScope.launch {
```

Comment:
i know this was here already, but launching a new coroutine here will likely mean a race condition, where we might finish deleting things before we restart the process, or maybe not.

the only function that calls `deleteAll()` is also a `suspend` function. what about if we made `deleteAll` also a `suspend` function, meaning you wouldn't need to `launch` a new coroutine? can you see any downsides?

Reply:
Sounds good to me, only called from a suspend function so that makes sense, changed!